packages/utils/src/conversions.ts

```diff
@@ -36,50 +38,60 @@ export function bigIntToHexadecimal(n: bigint): string {
 /**
  * Converts a hexadecimal string to a bigint. The input is interpreted as hexadecimal
  * with or without a '0x' prefix. It uses big-endian byte order.
- * @param hex The hexadecimal string to convert.
+ * @param value The hexadecimal string to convert.
  * @returns The bigint representation of the hexadecimal string.
  */
-export function beHexadecimalToBigInt(hex: string): bigint {
+export function beHexadecimalToBigInt(value: string): bigint {
+    if (!isHexadecimal(value) && !isHexadecimal(value, false)) {
+        throw new TypeError(`Parameter 'value' is not a hexadecimal string`)
+    }
+
     // Ensure the hex string starts with '0x'.
-    const formattedHexString = hex.startsWith("0x") ? hex : `0x${hex}`
+    const formattedHexString = value.startsWith("0x") ? value : `0x${value}`
 
     return BigInt(formattedHexString)
 }
 
 /**
  * Converts a hexadecimal string to a bigint. Alias for beHexadecimalToBigInt.
- * @param hex The hexadecimal string to convert.
+ * @param value The hexadecimal string to convert.
  * @returns The bigint representation of the hexadecimal string.
  */
-export function hexadecimalToBigInt(hex: string): bigint {
-    return beHexadecimalToBigInt(hex)
+export function hexadecimalToBigInt(value: string): bigint {
+    return beHexadecimalToBigInt(value)
 }
 
 /**
  * Converts a buffer of bytes to a bigint using big-endian byte order.
- * @param b The buffer to convert.
+ * It accepts 'Buffer' or 'Uint8Array'.
+ * @param value The buffer to convert.
  * @returns The bigint representation of the buffer's contents.
  */
-export function beBufferToBigInt(b: Buffer): bigint {
-    return BigInt(`0x${b.toString("hex")}`)
+export function beBufferToBigInt(value: Buffer | Uint8Array): bigint {
+    requireTypes(value, "value", ["Buffer", "Uint8Array"])
+
+    return BigInt(`0x${Buffer.from(value).toString("hex")}`)
 }
 
 /**
  * Converts a buffer to a bigint using little-endian byte order.
- * @param buffer The buffer to convert.
+ * It accepts 'Buffer' or 'Uint8Array'.
+ * @param value The buffer to convert.
  * @returns The bigint representation of the buffer's contents in little-endian.
  */
-export function leBufferToBigInt(buffer: Buffer): bigint {
-    return BigInt(`0x${Buffer.from(buffer).reverse().toString("hex")}`)
+export function leBufferToBigInt(value: Buffer | Uint8Array): bigint {
+    requireTypes(value, "value", ["Buffer", "Uint8Array"])
+
+    return BigInt(`0x${Buffer.from(value).reverse().toString("hex")}`)
 }
 
 /**
  * Converts a buffer to a bigint. Alias for beBufferToBigInt.
- * @param b The buffer to convert.
+ * @param value The buffer to convert.
  * @returns The bigint representation of the buffer's contents.
  */
-export function bufferToBigInt(b: Buffer): bigint {
-    return beBufferToBigInt(b)
+export function bufferToBigInt(value: Buffer | Uint8Array): bigint {
```

Comment:
I think Buffer is a subtype of Uint8Array, so the use of 2 types here is redundant, though it might be a good idea anyway for clarity.

Reply:
Yes, Buffer extends Uint8Array, but I'd keep it for clarity as you said.